src/DocumentFormat.OpenXml/Packaging/OpenXmlPackage.FlatOpc.cs

```diff
@@ -49,71 +56,89 @@ protected XDocument ToFlatOpcDocument(XProcessingInstruction instruction)
-                        new XAttribute(pkg + "contentType", part.ContentType),
-                        new XElement(pkg + "xmlData", XElement.Load(xmlReader)));
+                        Pkg + "part",
+                        new XAttribute(Pkg + "name", part.Uri),
+                        new XAttribute(Pkg + "contentType", part.ContentType),
+                        new XElement(Pkg + "xmlData", XElement.Load(xmlReader)));
                 }
             }
-            else
-            {
-                using (Stream stream = part.GetStream())
-                using (BinaryReader binaryReader = new BinaryReader(stream))
-                {
-                    int len = (int)binaryReader.BaseStream.Length;
-                    byte[] byteArray = binaryReader.ReadBytes(len);
 
-                    // The following expression creates the base64String, then chunks
-                    // it to lines of 76 characters long.
-                    string base64String = System.Convert.ToBase64String(byteArray)
-                        .Select((c, i) => new { Character = c, Chunk = i / 76 })
-                        .GroupBy(c => c.Chunk)
-                        .Aggregate(
-                            new StringBuilder(),
-                            (s, i) =>
-                                s.Append(
-                                    i.Aggregate(
-                                        new StringBuilder(),
-                                        (seed, it) => seed.Append(it.Character),
-                                        sb => sb.ToString())).Append(Environment.NewLine),
-                            s => s.ToString());
+            return GetBinaryPartContentsAsXml(part);
+        }
 
-                    return new XElement(
-                        pkg + "part",
-                        new XAttribute(pkg + "name", part.Uri),
-                        new XAttribute(pkg + "contentType", part.ContentType),
-                        new XAttribute(pkg + "compression", "store"),
-                        new XElement(pkg + "binaryData", base64String));
-                }
+        private static XElement GetBinaryPartContentsAsXml(PackagePart part)
+        {
+            return new XElement(
+                Pkg + "part",
+                new XAttribute(Pkg + "name", part.Uri),
+                new XAttribute(Pkg + "contentType", part.ContentType),
+                new XAttribute(Pkg + "compression", "store"),
+                new XElement(Pkg + "binaryData", ToChunkedBase64String(part)));
+        }
+
+        private static string ToChunkedBase64String(PackagePart part)
+        {
+            using (Stream stream = part.GetStream())
+            {
+                var byteArray = new byte[stream.Length];
+                stream.Read(byteArray, 0, byteArray.Length);
+                return ToChunkedBase64String(byteArray);
             }
         }
 
+        private static string ToChunkedBase64String(byte[] byteArray)
```

Comment:
I swear I've seen something like this in the code base before. Do you know if it's done else where? It may be useful if so to make it a general purpose utility

Reply:
I looked around but could not find anything else. You might have seen it in the Flat OPC code before. I turned it into a function, which I then beautified (e.g., formatted, renamed parameters).

Reply:
sounds good.